test/gtest/ucp/test_ucp_proto_mock.cc

```diff
@@ -337,6 +338,67 @@ class test_ucp_proto_mock : public ucp_test, public mock_iface {
         CMP_FIELD(mem_type);
         return true;
     }
+
+    void send_recv_am(size_t size)
+    {
+        /* Prepare receiver data handler */
+        mem_buffer recv_buf(size, UCS_MEMORY_TYPE_HOST);
+        struct ctx_t {
+            mem_buffer                     *buf;
+            bool                            received;
+            ucp_worker_h                    worker;
+            ucp_am_recv_data_nbx_callback_t cmpl;
+        } ctx = {&recv_buf, false, receiver().worker()};
+
+        ctx.cmpl = [](void *req, ucs_status_t status, size_t len, void *arg) {
+            ((ctx_t *)arg)->received = true;
+            ucp_request_free(req);
+        };
+
+        auto cb = [](void *arg, const void *header, size_t header_length,
+                     void *data, size_t len, const ucp_am_recv_param_t *param) {
+            ctx_t *ctx = (ctx_t *)arg;
+
+            if (param->recv_attr & UCP_AM_RECV_ATTR_FLAG_DATA) {
+                memcpy(ctx->buf->ptr(), data, len);
+                ctx->received = true;
+                return UCS_OK;
+            }
+
+            ucp_request_param_t params;
+            params.op_attr_mask = UCP_OP_ATTR_FIELD_CALLBACK |
+                                  UCP_OP_ATTR_FIELD_USER_DATA;
+            params.user_data    = arg;
+            params.cb.recv_am   = ctx->cmpl;
+
+            auto sptr = ucp_am_recv_data_nbx(ctx->worker, data, ctx->buf->ptr(),
+                                             len, &params);
+            EXPECT_FALSE(UCS_PTR_IS_ERR(sptr));
```

Comment:
need to make sure it is RNDV. Otherwise, if it is inline message, calling `ucp_am_recv_data_nbx` is invalid